### ISSUE TITLE
`mongodbatlas_online_archive` `schedule` parameter update causing crashing in `terraform apply`

### ISSUE THREAD

**Issue by neogibson** [mongodb/terraform-provider-mongodbatlas#1318]
### Terraform CLI and Terraform MongoDB Atlas Provider Version

```hcl
Terraform v1.4.4
MongoDB Atlas provider 1.10.1
```

### Terraform Configuration File 

```hcl
resource "mongodbatlas_online_archive" "notifications_archive" {
  project_id   = module.project.id
  cluster_name = "core-${var.environment}"
  coll_name    = "notifications"
  db_name      = "name"

  criteria {
    type              = "DATE"
    date_field        = "updatedAt"
    date_format       = "ISODATE"
    expire_after_days = 390
  }

  partition_fields {
    field_name = "userId"
    order      = 0
  }

  partition_fields {
    field_name = "type"
    order      = 1
  }

  lifecycle {
    ignore_changes = [
      criteria[0].expire_after_days,
      coll_name 
    ]
    prevent_destroy = true
  }
}
```

### Steps to Reproduce

1. Our CI Pipeline runs a `terraform plan`, resulting in the following plan:

![image](https://github.com/mongodb/terraform-provider-mongodbatlas/assets/79111296/7da24a30-75e9-4cf7-b766-a01ae49f8db0)

2. The next step in the CI pipeline attempts to run `terraform apply`

### Expected Behavior

- Terraform applies the change

### Actual Behavior

- The mongo atlas provider seems to crash, and subsequent `terraform plans` show the same change as above.

### Crash Output

```hcl
Error: configuring Terraform AWS Provider: validating provider credentials: retrieving caller identity from STS: operation error STS: GetCallerIdentity, https response error StatusCode: 0, RequestID: , request send failed, Post i/o timeout

The plugin encountered an error, and failed to respond to the
plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain
more details.

Error: Plugin did not respond

The plugin encountered an error, and failed to respond to the
plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain
more details.

Stack trace from the terraform-provider-mongodbatlas_v1.10.1 plugin:

panic: runtime error: index out of range [0] with length 0

goroutine 14 [running]:
github.com/mongodb/terraform-provider-mongodbatlas/mongodbatlas.mapSchedule(0xf9f8a0?)
    github.com/mongodb/terraform-provider-mongodbatlas/mongodbatlas/resource_mongodbatlas_online_archive.go:548 +0x35e
github.com/mongodb/terraform-provider-mongodbatlas/mongodbatlas.resourceMongoDBAtlasOnlineArchiveUpdate({0x14284a8, 0xc000cbed50}, 0xc000c4ed00, {0xf8cf40?, 0xc00072bb10?})
    github.com/mongodb/terraform-provider-mongodbatlas/mongodbatlas/resource_mongodbatlas_online_archive.go:419 +0x3bb
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0xc00063afc0, {0x14284e0, 0xc000c5d7d0}, 0xd?, {0xf8cf40, 0xc00072bb10})
    github.com/hashicorp/terraform-plugin-sdk/v2@v2.27.0/helper/schema/resource.go:767 +0x12e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc00063afc0, {0x14284e0, 0xc000c5d7d0}, 0xc000c1f520, 0xc000c4ea80, {0xf8cf40, 0xc00072bb10})
    github.com/hashicorp/terraform-plugin-sdk/v2@v2.27.0/helper/schema/resource.go:874 +0x845
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc0000e3c50, {0x14284e0?, 0xc000c5d6b0?}, 0xc000c6c000)
    github.com/hashicorp/terraform-plugin-sdk/v2@v2.27.0/helper/schema/grpc_provider.go:1024 +0xe8d
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc0002586e0, {0x14284e0?, 0xc000c5ccc0?}, 0xc000c3c150)
    github.com/hashicorp/terraform-plugin-go@v0.16.0/tfprotov5/tf5server/server.go:821 +0x574
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x108dbe0?, 0xc0002586e0}, {0x14284e0, 0xc000c5ccc0}, 0xc000c3c0e0, 0x0)
    github.com/hashicorp/terraform-plugin-go@v0.16.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:422 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0005481e0, {0x142d020, 0xc0006901a0}, 0xc000c5efc0, 0xc000647560, 0x1af1ba0, 0x0)
    google.golang.org/grpc@v1.56.0/server.go:1337 +0xdf3
google.golang.org/grpc.(*Server).handleStream(0xc0005481e0, {0x142d020, 0xc0006901a0}, 0xc000c5efc0, 0x0)
    google.golang.org/grpc@v1.56.0/server.go:1714 +0xa36
google.golang.org/grpc.(*Server).serveStreams.func1.1()
    google.golang.org/grpc@v1.56.0/server.go:959 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
    google.golang.org/grpc@v1.56.0/server.go:957 +0x18c

Error: The terraform-provider-mongodbatlas_v1.10.1 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

### Additional Context

- This only seemed to start happening after release of version `1.10.1` of the provider.

**Comment by andreaangiolillo** [comment]
Hi @neogibson.

Thanks for opening the issue. Yes, this is a bug of 1.10.1. I am working on the fix.

In the meantime, the workaround is to add `schedule.type="DEFAULT"` to your `mongodbatlas_online_archive` resource. Keep in mind that `schedule.type="DEFAULT"` is the value set to the online archive resource in MongoDB Atlas when the `schedule` param is not provided during resource creation.
```hcl
  schedule {
      type         = "DEFAULT"
  } 
```

This is your config file with the `schedule` param:
```hcl
resource "mongodbatlas_online_archive" "notifications_archive" {
  project_id   = module.project.id
  cluster_name = "core-${var.environment}"
  coll_name    = "notifications"
  db_name      = "name"

  criteria {
    type              = "DATE"
    date_field        = "updatedAt"
    date_format       = "ISODATE"
    expire_after_days = 390
  }

  schedule {
      type         = "DEFAULT"
  } 

  partition_fields {
    field_name = "userId"
    order      = 0
  }

  partition_fields {
    field_name = "type"
    order      = 1
  }

  lifecycle {
    ignore_changes = [
      criteria[0].expire_after_days,
      coll_name 
    ]
    prevent_destroy = true
  }
}
```

Thanks

**Comment by neogibson** [comment]
Thanks for the response! We will do the workaround for now 👍

**Comment by andreaangiolillo** [comment]
Hello @neogibson 
We released [v1.10.2](https://registry.terraform.io/providers/mongodb/mongodbatlas/1.10.2) with the fix to this bug. Make sure to remove the `schedule.type="DEFAULT"` field from your configuration file before upgrading to 1.10.2

```terraform
// you need to remove this
  schedule {
      type         = "DEFAULT"
  }
```

Thanks

**Comment by lynnporu** [comment]
This error happened to me when I was trying to import an `mongodbatlas_advanced_cluster` object. It turned out that I used wrong cluster name while importing (`terraform import mongodb_advanced_cluster.cluster-name PROJECT_ID-WRONG-CLUSTER_NAME`). So check your commands twice.
